tests/integration/synapseclient/integration_test.py

```diff
@@ -870,3 +870,293 @@ def test_get_permissions_for_project_with_public_and_registered_user(self):
+            Project(
+                name=str(uuid.uuid4())
+                + "test_get_permissions_through_two_teams_assigned_to_user_and_project"
+            )
+        )
+
+        # AND the user that created the project
+        p1: UserProfile = self.syn.getUserProfile()
+
+        # AND a team is created
+        name = "My Uniquely Named Team " + str(uuid.uuid4())
+        team_1 = self.syn.store(
+            Team(
+                name=name,
+                description="A fake team for testing permissions assigned to a single project - 1",
+            )
+        )
+
+        # AND a second team is created
+        name = "My Uniquely Named Team " + str(uuid.uuid4())
+        team_2 = self.syn.store(
+            Team(
+                name=name,
+                description="A fake team for testing permissions assigned to a single project - 2",
+            )
+        )
+
+        # Handle Cleanup - Note: When running this schedule for cleanup order can matter when there are dependent resources
+        self.schedule_for_cleanup(team_1)
+        self.schedule_for_cleanup(team_2)
+        self.schedule_for_cleanup(project_with_permissions_through_multiple_teams)
+
+        # AND the permissions for the Team 1 on the entity are set to all permissions except for DOWNLOAD
+        self.syn.setPermissions(
+            project_with_permissions_through_multiple_teams,
+            team_1.id,
+            [
+                "READ",
+                "DELETE",
+                "CHANGE_SETTINGS",
+                "UPDATE",
+                "CHANGE_PERMISSIONS",
+                "CREATE",
+                "MODERATE",
+            ],
+        )
+
+        # AND the permissions for the Team 2 on the entity are set to only READ and DOWNLOAD
+        self.syn.setPermissions(
+            project_with_permissions_through_multiple_teams,
+            team_2.id,
+            ["READ", "DOWNLOAD"],
+        )
+
+        # AND the permissions for the user on the entity are set to NONE
+        self.syn.setPermissions(
+            project_with_permissions_through_multiple_teams, p1.ownerId, []
+        )
+
+        # WHEN I get the permissions for the user on the entity
+        permissions = self.syn.get_permissions(
+            project_with_permissions_through_multiple_teams.id
+        )
+
+        # THEN I expect to see the permissions of both teams
+        expected_permissions = [
+            "READ",
+            "DELETE",
+            "CHANGE_SETTINGS",
+            "UPDATE",
+            "CHANGE_PERMISSIONS",
+            "CREATE",
+            "MODERATE",
+            "DOWNLOAD",
+        ]
+        assert set(expected_permissions) == set(permissions.access_types)
+
+    @tracer.start_as_current_span(
+        "integration_test::TestPermissionsOnEntityForCaller::test_get_permissions_for_project_with_public_and_registered_user"
+    )
+    def test_get_permissions_for_project_with_public_and_registered_user(self):
```

Comment:
Is this test testing what we want out of it given that `get_permissions` only grabs the current logged in user permissions on an entity? Are we still able to get what the public user id: `273949` has as far as permissions?